AbstractPostListViewController: Crash on launch

Just experienced the app crashing on launch whilst working on another issue. Looks like it's to do with state restoration, based on the stack trace. I'm afraid I don't have any repro steps at the moment, but I'll update if I find a way to trigger it.

```
* thread #1: tid = 0x3be756, 0x00000001024e6588 libswiftCore.dylib`function signature specialization <Arg[0] = Exploded, Arg[1] = Exploded, Arg[2] = Dead, Arg[3] = Dead> of Swift._fatalErrorMessage (Swift.StaticString, Swift.StaticString, Swift.StaticString, Swift.UInt) -> () + 44, stop reason = EXC_BREAKPOINT (code=1, subcode=0x1024e6588)
    frame #0: 0x00000001024e6588 libswiftCore.dylib`function signature specialization <Arg[0] = Exploded, Arg[1] = Exploded, Arg[2] = Dead, Arg[3] = Dead> of Swift._fatalErrorMessage (Swift.StaticString, Swift.StaticString, Swift.StaticString, Swift.UInt) -> () + 44
  * frame #1: 0x0000000100604f4c WordPress`AbstractPostListViewController.canFilterByAuthor(self=0x0000000137819b70) -> Bool + 104 at AbstractPostListViewController.swift:924
    frame #2: 0x00000001005b88cc WordPress`PostListViewController.currentPostAuthorFilter(self=0x0000000137819b70) -> AbstractPostListViewController.PostAuthorFilter + 72 at PostListViewController.swift:563
    frame #3: 0x0000000100605310 WordPress`AbstractPostListViewController.availablePostListFilters(self=<unavailable>) -> [PostListFilter] + 52 at AbstractPostListViewController.swift:942
    frame #4: 0x00000001006057d0 WordPress`AbstractPostListViewController.currentPostListFilter(self=<unavailable>) -> PostListFilter + 48 at AbstractPostListViewController.swift:953
    frame #5: 0x0000000100606844 WordPress`AbstractPostListViewController.updateFilterTitle(self=<unavailable>) -> () + 168 at AbstractPostListViewController.swift:1027
    frame #6: 0x00000001005f9340 WordPress`AbstractPostListViewController.configureNavbar(self=<unavailable>) -> () + 776 at AbstractPostListViewController.swift:194
    frame #7: 0x00000001005f80c0 WordPress`AbstractPostListViewController.viewDidLoad(self=<unavailable>) -> () + 380 at AbstractPostListViewController.swift:102
    frame #8: 0x00000001005af714 WordPress`PostListViewController.viewDidLoad(self=<unavailable>) -> () + 48 at PostListViewController.swift:98
    frame #9: 0x00000001005af8dc WordPress`@objc PostListViewController.viewDidLoad() -> () + 40 at PostListViewController.swift:0
    frame #10: 0x0000000185f9cc40 UIKit`-[UIViewController loadViewIfRequired] + 996
    frame #11: 0x0000000185f9c844 UIKit`-[UIViewController view] + 28
    frame #12: 0x000000018611dc00 UIKit`-[UIApplication(StateRestoration) _restoreApplicationPreservationStateWithSessionIdentifier:beginHandler:completionHandler:] + 5172
    frame #13: 0x0000000186012d44 UIKit`-[UIApplication(StateRestoration) _doRestorationIfNecessary] + 244
    frame #14: 0x0000000186012964 UIKit`-[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 308
    frame #15: 0x0000000186242184 UIKit`-[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 2904
    frame #16: 0x00000001862465f0 UIKit`-[UIApplication _runWithMainScene:transitionContext:completion:] + 1684
    frame #17: 0x0000000186243764 UIKit`-[UIApplication workspaceDidEndTransaction:] + 168
    frame #18: 0x00000001827df7ac FrontBoardServices`__FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 36
    frame #19: 0x00000001827df618 FrontBoardServices`-[FBSSerialQueue _performNext] + 168
    frame #20: 0x00000001827df9c8 FrontBoardServices`-[FBSSerialQueue _performNextFromRunLoopSource] + 56
    frame #21: 0x0000000180df509c CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
    frame #22: 0x0000000180df4b30 CoreFoundation`__CFRunLoopDoSources0 + 540
    frame #23: 0x0000000180df2830 CoreFoundation`__CFRunLoopRun + 724
    frame #24: 0x0000000180d1cc50 CoreFoundation`CFRunLoopRunSpecific + 384
    frame #25: 0x000000018600b94c UIKit`-[UIApplication _run] + 460
    frame #26: 0x0000000186006088 UIKit`UIApplicationMain + 204
    frame #27: 0x00000001000752dc WordPress`main(argc=<unavailable>, argv=<unavailable>) + 84 at main.m:9
    frame #28: 0x00000001808ba8b8 libdyld.dylib`start + 4
```

Tested on `issue/5599-modal-presentation-styles`, iPhone 6s running 9.3.2

I just confirmed my suspicion  above... the problem is that `blog` is `nil`, although I'm not yet sure how that's possible.

<img width="1169" alt="screen shot 2016-07-18 at 12 19 22 pm" src="https://cloud.githubusercontent.com/assets/1836005/16919960/13ca6e42-4ce2-11e6-8d23-4a335bce359b.png">

```
(lldb) po blog
fatal error: unexpectedly found nil while unwrapping an Optional value
expression produced error: /var/folders/wn/zn1s4q_d5pggv806t6st86s40000gn/T/lldb/596/expr8.swift:1:80: error: use of undeclared type '__ObjC'
$__lldb__DumpForDebugger(Swift.UnsafePointer<Swift.ImplicitlyUnwrappedOptional<__ObjC.Blog>>(bitPattern: 0x1239d7ff0).memory)
                                                                               ^~~~~~
/var/folders/wn/zn1s4q_d5pggv806t6st86s40000gn/T/lldb/596/expr8.swift:1:45: note: while parsing this '<' as a type parameter bracket
$__lldb__DumpForDebugger(Swift.UnsafePointer<Swift.ImplicitlyUnwrappedOptional<__ObjC.Blog>>(bitPattern: 0x1239d7ff0).memory)
                                            ^
```

Reproduction steps:
1. Launch the app from `release/6.2`.
2. Send the app to background, and stop execution from Xcode's stop button. (IMPORTANT STEP)
3. Launch the app from `release/6.3`.
4. Crash!

--- a/WordPress/Classes/System/WordPressAppDelegate.m
+++ b/WordPress/Classes/System/WordPressAppDelegate.m
@@ -354,7 +354,20 @@ int ddLogLevel = DDLogLevelInfo;
 
 - (BOOL)application:(UIApplication *)application shouldRestoreApplicationState:(NSCoder *)coder
 {
-    return self.shouldRestoreApplicationState;
+    NSUserDefaults* standardUserDefaults = [NSUserDefaults standardUserDefaults];
+
+    NSString* const lastSavedStateVersionKey = @"lastSavedStateVersionKey";
+    NSString* lastSavedStateVersion = [standardUserDefaults objectForKey:lastSavedStateVersionKey];
+    NSString* currentVersion = [[NSBundle mainBundle] objectForInfoDictionaryKey: @"CFBundleShortVersionString"];
+    BOOL shouldRestoreApplicationState = NO;
+
+    if (lastSavedStateVersion && [lastSavedStateVersion length] > 0 && [lastSavedStateVersion isEqualToString:currentVersion]) {
+        shouldRestoreApplicationState = self.shouldRestoreApplicationState;;
+    }
+
+    [standardUserDefaults setObject:currentVersion forKey:lastSavedStateVersionKey];
+
+    return shouldRestoreApplicationState;
 }
 
 - (void)application: (UIApplication *)application performActionForShortcutItem:(nonnull UIApplicationShortcutItem *)shortcutItem completionHandler:(nonnull void (^)(BOOL))completionHandler

